pkg/resources/envoy/configmap.go

```diff
@@ -35,15 +36,41 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-func (r *Reconciler) configMap(log logr.Logger) runtime.Object {
+func configName(envoyConfig *v1beta1.EnvoyConfig) string {
+	if envoyConfig.Id == envoyGlobal {
+		return envoyVolumeAndConfigName
+	} else {
+		return fmt.Sprintf("%s-%s", envoyVolumeAndConfigName, envoyConfig.Id)
+	}
+}
+
+func (r *Reconciler) configMap(log logr.Logger, envoyConfig *v1beta1.EnvoyConfig) runtime.Object {
 	configMap := &corev1.ConfigMap{
-		ObjectMeta: templates.ObjectMeta(envoyVolumeAndConfigName, labelSelector, r.KafkaCluster),
-		Data:       map[string]string{"envoy.yaml": generateEnvoyConfig(r.KafkaCluster, log)},
+		ObjectMeta: templates.ObjectMeta(configName(envoyConfig), labelSelector(envoyConfig), r.KafkaCluster),
+		Data:       map[string]string{"envoy.yaml": generateEnvoyConfig(r.KafkaCluster, envoyConfig, log)},
 	}
 	return configMap
 }
 
-func generateEnvoyConfig(kc *v1beta1.KafkaCluster, log logr.Logger) string {
+func firstExternalListener(kc *v1beta1.KafkaCluster, broker v1beta1.Broker) v1beta1.ExternalListenerConfig {
```

Comment:
can we actually have more external listners?

Reply:
We should be able to have multiple listeners. This was broken / hard-coded. I will do some testes to see how it behaves now.

Reply:
I did some tests: we can have multiple external listeners now. The only problem is that Envoy can use only 1 external listener. It will use the first from the list. However, if we could add multiple ingresses (or an ingress that is not manged by the Operator), we could use multiple external listeners. They are all created on the broker and Kafka listen to all of them now.